File: dotnet/Lexy.Poc/Lexy.Poc.Core/Parser/Tokens/StringLiteralToken.cs

namespace Lexy.Poc.Core.Parser.Tokens
{
    public class StringLiteralToken : Token, ILiteralToken
    {
        public override string Value { get; }

        public StringLiteralToken(string value)
        {
            Value = value;
        }

        public override string ToString() => Value;
    }
}

File: typescipt/src/parser/tokens/stringLiteralToken.ts


export class StringLiteralToken extends Token, ILiteralToken {
   public StringLiteralToken(string value, TokenCharacter character) : base(character) {
     Value = value;
   }

   public override string Value

   public object TypedValue => Value;

   public deriveType(context: IValidationContext): VariableType {
     throw new Error(`Not supported. Type should be defined by node or expression.`);
   }

   public override toString(): string {
     return Value;
   }
}
